### DISCUSSION TITLE
Webui evolution - rely on web front framework

### DISCUSSION THREAD

**Discussion by klevain** [transmission/transmission#5489] [Show and tell]
The more I play with the webui interface the more I realise that the webui can't be the exact replica of qt desktop application. I would rather have a strong UI system based on the medium I dev for (qt for desktop) than trying to rebuild it on a medium that maybe can't take it (currently web UI).

With that in mind I've made some tests around mainly with [bootstrap](https://getbootstrap.com/docs/5.3) and [tailwind](https://tailwindcss.com/).

I've easily come with a really [neat preview with bootstrap](https://codepen.io/klevain/full/RwewoQJ) and [tailwind is more difficult](https://play.tailwindcss.com/PvdwJBCLtQ) to implement without deep JS change (template tool / or component based frameworks : vuejs, react...).

What it does and why it feels good :
- it fixes a lot off CSS related issue : color contrast, spacing...
- it helps a lot building template, either adapt from example or direct implementation.
- it inherently creates uniformity between elements

Wasn't sure how to proceed here so I've publish my try in [this branch on my fork](https://github.com/klevain/transmission/tree/feature/bootstrapified). It's still totally work in progress but bootstrap is available (and I really like that).

 ⚠️
- currently just the navbar has been bsified.
- The css is still in the middle so some visual bugs can appear all along.
- It's not optimized (the css in the middle seems to bloat the webpack compilation)

So here I ping "people of interest" : @ckerr, @GaryElshaw, @dareiff, @S-Aarab  to discuss this option.

Please give feedbacks 😄

**Comment by GaryElshaw** [comment]
The current webui is so new, i'm not sure how i feel about a major refactor, right now (as much as i like the aesthetic of your work). My own personal project priorities are about getting the below merged so that the qt client looks modernised, and coincidentally, it makes the webui look like it is a part of the same family. Ultimately, and personally, I'd like for users to be able to go from the ui of whatever client they use to a web ui that looks and feels seamless from their chosen client.

![image](https://user-images.githubusercontent.com/69029666/216804725-cc695f76-4a0b-4bb5-b929-549bb4c898a6.png) 

In that vein, and it's probably outside interest of the project(?), i'm going to create a set of macOS icons that users can integrate into the web ui if they're using the macOS client.

**Comment by dareiff** [comment]
FWIW, I started down a new framework path and realized how much (really good) work exists in the current webui, so I abandoned it. I think there's a lot of good that could come from a more ... functional web interface, but the groundwork that exists in its current form is pretty significant.

And now that I've written that, I'm realizing you're looking for more of a css framework. In THAT case, I'd suggest solidifying the work already there in a set of definable components (buttons equal this, links equal this, etc.).
